monkey/monkey_island/cc/app.py

```diff
@@ -72,36 +73,54 @@ def serve_home():
     return serve_static_file(HOME_FILE)
 
 
-def init_app_config(app, mongo_url, data_dir: Path):
+def setup_authentication(app, data_dir):
+    flask_security_config = generate_flask_security_configuration(data_dir)
+
+    # TODO: After we switch to token base authentication investigate the purpose
+    # of `SECRET_KEY` and `SECURITY_PASSWORD_SALT`, take into consideration
+    # the discussion https://github.com/guardicore/monkey/pull/3006#discussion_r1116944571
+    app.config["SECRET_KEY"] = flask_security_config["secret_key"]
+    app.config["SECURITY_PASSWORD_SALT"] = flask_security_config["password_salt"]
+    app.config["SECURITY_USERNAME_ENABLE"] = True
+    app.config["SECURITY_USERNAME_REQUIRED"] = True
+    app.config["SECURITY_REGISTERABLE"] = True
+    app.config["WTF_CSRF_CHECK_DEFAULT"] = False
+    app.config["SECURITY_CSRF_IGNORE_UNAUTH_ENDPOINTS"] = True
+    app.config["SECURITY_SEND_REGISTER_EMAIL"] = False
+
+    # The database object needs to be created after we configure the flask application
     db = MongoEngine(app)
+
+    class CustomConfirmRegisterForm(ConfirmRegisterForm):
+        # Flask-Security doesn't let you register without an email field
+        email = StringField("Email", default="dummy@dummy.com")
+        password = PasswordField(
+            "Password",
+            [validators.DataRequired(), validators.Length(min=1, max=25)],
+        )
```

Comment:
Why are we overriding the password field?

Reply:
Validators... While we test we will need to enter passwords with above 8 chars. We can add comment to delete it.

Reply:
![image](https://user-images.githubusercontent.com/36815064/222136535-9d2bfe62-c685-464b-a0de-d329535d2e79.png)